atd-etl/app/process/helpers_hasura_geocode.py

```diff
@@ -0,0 +1,413 @@
+    """
+    Returns true if the street contains any of the below strings
+    :param street: string - The name of the string
+    :return: bool
+    """
+    # If the street is empty, then it is faulty.
+    if street == "" or street is None:
+        return True
+
+    list_of_bad_keywords = [
+        "PARKING LOT",
+        "NOT REPORTED",
+        "PRIVATE ROAD",
+    ]
+
+    # If it contains the words NOT REPORTED
+    for bad_keyword in list_of_bad_keywords:
+        if bad_keyword in street:
+            return True
+
+    return False
+
+
+def address_has_numbers(address):
+    """
+    Returns true if an address has a block number
+    :param address: string - The address being evaluated
+    :return: bool
+    """
+    return any(char.isdigit() for char in address)
+
+
+def both_block_num_missing(record):
+    """
+    Returns true of both block numbers are missing
+    :param record: dict - The record being evaluated
+    :return: bool
+    """
+    rpt_block_num = record.get("rpt_block_num", "") or ""
+    rpt_sec_block_num = record.get("rpt_sec_block_num", "") or ""
+
+    # True, if neither address has a block number.
+    if rpt_block_num == "" and rpt_sec_block_num == "":
+        return True
+
+    return False
+
+
+def is_intersection(record):
+    """
+    Returns True if both streets have no block number.
+    :param record: dict - The record being evaluated
+    :return: bool
+    """
+    rpt_street_name = record.get("rpt_street_name", "") or ""
+    rpt_sec_street_name = record.get("rpt_sec_street_name", "") or ""
+
+    # If either street name is empty, return false
+    if rpt_street_name == "" and rpt_sec_street_name == "":
+        return False
+
+    # True, if neither address has a block number.
+    if both_block_num_missing(record):
+        return True
+
+    return False
+
+
+def get_match_quality_here(response):
+    """
+    Returns the quality level from 0 to 1
+    :param response: dict - The full response from Here
+    :return: float
+    """
+
+    try:
+        return response["Response"]["View"][0]["Result"][0]["Relevance"]
+    except:
+        return 0
+
+
+def geocode_address_here(address):
+    """
+    Runs a geocode request against the Here API
+    :param address: string - The address to pass to the Here endpoint.
+    :return: string
+    """
+
+    # Build our parameters
+    parameters = {
+        "gen": "9",
+        "prox": "30.268064,-97.742814,1000",
+        "app_id": ATD_ETL_CONFIG["ATD_HERE_APP_ID"],
+        "app_code": ATD_ETL_CONFIG["ATD_HERE_APP_CODE"],
+        "mapview": ATD_ETL_CONFIG["ATD_HERE_BOUNDING_BOX"],
+        "searchtext": address,
+    }
+
+    try:
+        # Make request to API Endpoint
+        return requests.get(
+            ATD_ETL_CONFIG["ATD_HERE_API_ENDPOINT"], params=parameters
+        ).json()
+        # coordinates = request.json()['Response']['View'][0]['Result'][0]['Location']['DisplayPosition']
+    except Exception as e:
```

Comment:
I know we talked about this earlier, but this is where the error msg `{"error": "Expecting value: line 1 column 1 (char 0)"}` comes from, right? Just want to make sure I understand the log.

Reply:
Correct, the .json method assumes there is a valid response object. If not it will raise an exception which gets caught as an error message in the lines below.